### PR TITLE
Add a new GitHub action for contribution PRs

### PR DESCRIPTION
**Pull request by RotemAmit**
## Status
In Progress

## Related Issues
Related: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8750)

## Description
Added a new Github action that will assign the contribution TL to the contributions PRs and will add the Contribution label.

### CODE REVIEW

**Comment by RotemAmit** [comment]
A test [PR](https://github.com/demisto/dockerfiles/pull/21579).

File: .github/workflows/handle-new-contribution.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Add an assignee and a Contribution label
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+          PR_URL: ${{github.event.pull_request.html_url}}
+        run: |
+          echo "get content_roles.json using curl"
+          content_roles=$( curl https://raw.githubusercontent.com/demisto/content/master/.github/content_roles.json )
+          echo "$content_roles"
+          
+          echo "==== $(date): Starting pipenv setup... ===="
```
Comment (JudahSchwartz) [inline .github/workflows/handle-new-contribution.yml:38]: all of this seems like a lot of work for something simple. We dont need the python here, or the files
We can use jq for this

`curl theurl | jq .CONTRIBUTION_TL` will give us the tl name

File: .github/workflows/handle-new-contribution.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  assign_and_label:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.head.repo.fork == true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Setup Python
```
Comment (JudahSchwartz) [inline .github/workflows/handle-new-contribution.yml:20]: dont think we still need python

File: .github/workflows/handle-new-contribution.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  assign_and_label:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.head.repo.fork == true
+    steps:
+      - name: Checkout
```
Comment (JudahSchwartz) [inline .github/workflows/handle-new-contribution.yml:16]: do we actually need to checkout? We dont need the code from the repo

File: .github/workflows/handle-new-contribution.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+      - name: Print Context
```
Comment (JudahSchwartz) [inline .github/workflows/handle-new-contribution.yml:24]: is this for debugging? Do we need it?

File: .github/workflows/handle-new-contribution.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+          PR_URL: ${{github.event.pull_request.html_url}}
+        run: |
+          echo "get content_roles.json using curl"
+          contrib_tl=$( curl https://raw.githubusercontent.com/demisto/content/master/.github/content_roles.json | jq .CONTRIBUTION_TL -r )
+          echo "$contrib_tl"
+          echo "Finished retrieving the contribution TL username"
+          
+          if [ $contrib_tl ]; then
```
Comment (JudahSchwartz) [inline .github/workflows/handle-new-contribution.yml:39]: does this work? we dont need -z or -n?

File: .github/workflows/handle-new-contribution.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            echo "Add an assignee and a Contribution label"
+            gh pr edit --add-assignee "$contrib_tl" --add-label Contribution "$PR_URL"
```
Comment (JudahSchwartz) [inline .github/workflows/handle-new-contribution.yml:41]: I think we should also add reviewer. If i was the TL I would miss it otherwise

**Comment by JudahSchwartz** [comment]
When there's a new contribution from a contributor that hasnt contributed before, theres an "approve and run" button that we need to click before other github actions run. Does this run without needing to click it? it might be a setting somewhere but lets make sure this doesnt need manual intervention to run

File: .github/workflows/handle-new-contribution.yml (hunk 5 above)
Comment (RotemAmit) [reply .github/workflows/handle-new-contribution.yml:39]: yes, it works, I checked in the related test PR. Actually, when I tried with -z it didn't work. I took it from [here](https://serverfault.com/questions/7503/how-to-determine-if-a-bash-variable-is-empty).
image

**Comment by RotemAmit** [comment]
> When there's a new contribution from a contributor that hasnt contributed before, theres an "approve and run" button that we need to click before other github actions run. Does this run without needing to click it? it might be a setting somewhere but lets make sure this doesnt need manual intervention to run

It is working in this [PR](https://github.com/demisto/dockerfiles/pull/21624).

**Review by JudahSchwartz — APPROVED** [review]
great work!

**Pull request merged**